eox_nelp/api_clients/tests/__init__.py

```diff
@@ -177,3 +141,80 @@ def test_failed_get(self, auth_mock, requests_mock):
         self.assertEqual(logs.output, [
             f"ERROR:{api_clients.__name__}:{log_error}"
         ])
+
+
+class TestOauth2ApiClientMixin(TestApiClientMixin):
+    """
+    This test class contains test cases for the `AbstractOauth2ApiClient` class
+    to ensure that the authentication process using OAuth2 is working correctly.
+    """
+
+    def test_failed_authentication(self):
+        """Test case for invalid credentials.
+
+        Expected behavior:
+            - Raise MissingTokenError exception
+        """
+        self.assertRaises(MissingTokenError, self.api_class)
+
+    @patch("eox_nelp.api_clients.OAuth2Session")
+    def test_successful_authentication(self, oauth2_session_mock):
+        """Test case when the authentication response is valid.
+
+        Expected behavior:
+            - Session is set
+            - Session headers contains Authorization key.
+            - fetch_token was called with the right values.
+        """
+        fetch_token_mock = Mock()
+        fetch_token_mock.return_value = {
+            "token_type": "Bearer",
+            "access_token": "12345678abc",
+            "expires_in": 200,
+        }
+        oauth2_session_mock.return_value.fetch_token = fetch_token_mock
+        authentication_url = f"{settings.FUTUREX_API_URL}/oauth/token"
```

Comment:
```suggestion
        authentication_url = f"{settings.FUTUREX_API_URL}/oauth2/token"
```
?

Reply:
from the doc

![image](https://github.com/eduNEXT/eox-nelp/assets/36200299/a02edc3d-2d33-44e2-afa8-82eec0ee33bd)

https://drive.google.com/file/d/16PXRFCFZuRQMjleQqkm_qSKSfddpwT8g/view